src/library/dao/directorydao.h

```diff
@@ -28,8 +29,14 @@ class DirectoryDAO : public DAO {
     };
     RemoveResult removeDirectory(const mixxx::FileInfo& oldDir) const;
 
+    enum class RelocateResult {
+        Ok,
+        InvalidOrMissingDirectory,
+        UnreadableDirectory,
+        SqlError,
+    };
```

Comment:
To reduce tr strings for the dialogs we could merge all `~Result` types and forward them to a dialog function in Library.